### ISSUE TITLE
md-svg-icon attr value should function like ng-class

### ISSUE THREAD

**Issue by jeremymwells** [angular/material#4157]
It'd be awesome if md-svg-icon functioned like ng-class so as to be able to:

```html
<md-icon 
        md-svg-icon="{'action-foo': modelTrue, 'action-bar':!modelTrue}" 
        ng-click="modelTrue=!modelTrue">
</md-icon>
````

rather than:

```html
<md-icon 
        md-svg-icon="action-foo" ng-if="modelTrue" 
        ng-click="modelTrue=!modelTrue">
</md-icon>`

<md-icon 
        md-svg-icon="action-bar" ng-if="!modelTrue" 
        ng-click=modelTrue=!modelTrue">
</md-icon>
```

**Comment by jeremymwells** [comment]
@ThomasBurleson is there a reason for isolated scope on the md-icon directive? Or perhaps more relevant, is/are there reason(s) to not use inherited scope in this directive?

**Comment by ThomasBurleson** [comment]
@jeremymwells - this sounds like a very nice feature enhancement; to be addressed in post 1.0.

> As a best practice we try to use isolated scopes within our components.
